packages/graphql-typescript-definitions/src/filesystem/default-graphql-filesystem.ts

```diff
@@ -0,0 +1,74 @@
+import {FSWatcher, watch} from 'chokidar';
+import {GraphQLProjectConfig, GraphQLConfig} from 'graphql-config';
+import {
+  getGraphQLProjectIncludedFilePaths,
+  getGraphQLProjects,
+  getGraphQLSchemaPaths,
+  resolvePathRelativeToConfig,
+} from 'graphql-config-utilities';
+
+import {AbstractGraphQLFilesystem} from './graphql-filesystem';
+
+export class DefaultGraphQLFilesystem extends AbstractGraphQLFilesystem {
+  private readonly watchers: FSWatcher[] = [];
+
+  async watch(config: GraphQLConfig) {
+    this.watchers.push(
+      ...this.setupDocumentWatchers(config).concat(
+        this.setupSchemaWatcher(config),
+      ),
+    );
+
+    await Promise.all(
+      this.watchers.map(
+        watcher =>
+          new Promise<void>(resolve => watcher.on('ready', () => resolve())),
+      ),
+    );
+  }
+
+  dispose() {
+    this.watchers.forEach(watcher => {
+      watcher.close();
+    });
+
+    this.watchers.length = 0;
+  }
+
+  getGraphQLProjectIncludedFilePaths(projectConfig: GraphQLProjectConfig) {
+    return getGraphQLProjectIncludedFilePaths(projectConfig);
+  }
+
+  private setupDocumentWatchers(config: GraphQLConfig) {
```

Comment:
Adapting migrated code to trigger `change` / `delete` events https://github.com/Shopify/quilt/blob/6488657a0287d6b9de9e8811f623e92de5b7f92c/packages/graphql-typescript-definitions/src/index.ts#L207-L235